scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,336 @@
+        **model_kwargs,
+    ):
+        super().__init__(adata)
+
+        if prior == "vamp":
+            if pseudoinputs_data_indices is None:
+                pseudoinputs_data_indices = np.random.randint(
+                    0, adata.shape[0], n_prior_components
+                )
+            pseudoinput_data = next(
+                iter(
+                    self._make_data_loader(
+                        adata=adata,
+                        indices=pseudoinputs_data_indices,
+                        batch_size=n_prior_components,
+                        shuffle=False,
+                    )
+                )
+            )
+        else:
+            pseudoinput_data = None
+
+        n_cov_const = (
+            adata.obsm["covariates"].shape[1] if "covariates" in adata.obsm else 0
+        )
+        cov_embed_sizes = (
+            pd.DataFrame(adata.obsm["covariates_embed"]).nunique(axis=0).to_list()
+            if "covariates_embed" in adata.obsm
+            else []
+        )
+
+        # self.summary_stats provides information about anndata dimensions and other tensor info
+        self.module = SysVAE(
+            n_input=adata.shape[1],
+            n_cov_const=n_cov_const,
+            cov_embed_sizes=cov_embed_sizes,
+            n_system=adata.obsm["system"].shape[1],
+            prior=prior,
+            n_prior_components=n_prior_components,
+            pseudoinput_data=pseudoinput_data,
+            **model_kwargs,
+        )
+
+        self._model_summary_string = (
+            "cVAE model with optional VampPrior and latent cycle-consistency loss"
+        )
+        # necessary line to get params that will be used for saving/loading
+        self.init_params_ = self._get_init_params(locals())
+
+        logger.info("The model has been initialized")
+
+    @torch.no_grad()
+    def get_latent_representation(
+        self,
+        adata: AnnData,
+        indices: Sequence[int] | None = None,
+        cycle: bool = False,
+        give_mean: bool = True,
+        batch_size: int | None = None,
+        as_numpy: bool = True,
+    ) -> np.ndarray | torch.Tensor:
+        """Return the latent representation for each cell.
+
+        Parameters
+        ----------
+        adata
+            Input adata for which latent representation should be obtained.
+        indices
+            Data indices to embed. If None embedd all cells.
+        cycle
+            Return latent embedding of the cycle pass.
+        give_mean
+            Return the posterior mean instead of a sample from the posterior.
+        batch_size
+            Minibatch size for data loading into model. Defaults to `scvi.settings.batch_size`.
+        as_numpy
+            Return in numpy rather than torch format.
+
+        Returns
+        -------
+        Latent Embedding
+        """
+        # Check model and adata
+        self._check_if_trained(warn=False)
+        # TODO extend to check if adata setup is correct wrt training data
```

Comment:
What other checks would you like to implement here?

Reply:
I will re-implement _validate_anndata to do the checks that SysVI requires